src/Device/Driver/AR62xx.cpp

```diff
@@ -0,0 +1,573 @@
+      else
+        crc <<= 1;
+    }
+  }
+  return (crc);
+}
+
+/**
+   * @brief generates a command to set a station in the AR62xx
+   * 
+   * @param command command-output to set the station
+   * @param station_id the id of the station which should be set
+   *        1 = active station of AR62xx
+   *        0 = passive station of AR62xx
+   * @param frequency frequency in kHz to set
+   * @param station name of the station to set
+   *        (unused)
+   * @return int command-length
+   */
+int AR62xxDevice::GenerateSetStationCommand(uint8_t *command, int active_passive, RadioFrequency frequency, const TCHAR *station)
+{
+  /** length of the command */
+  unsigned int command_length = 0;
+
+  /** convert the active-frequency to an AR62xx-ID */
+  uint16_t active_frequency_idx = FrequencyToAR62xxId(active_frequency);
+
+  /** convert the passive-frequency to an AR62xx-ID */
+  uint16_t passive_frequency_idx = FrequencyToAR62xxId(passive_frequency);
+
+  /** command-start */
+  command[command_length++] = 0xA5;
+
+  /** add prot-ID */
+  command[command_length++] = 0x14;
+  command[command_length++] = 5;
+
+  /** set active station */
+  if (active_passive == ACTIVE_STATION)
+    active_frequency_idx = FrequencyToAR62xxId(frequency);
+
+  /** set passive station */
+  else
+    passive_frequency_idx = FrequencyToAR62xxId(frequency);
+
+  /** setting frequencies */
+  command[command_length++] = 22;
+  command[command_length++] = GetNthByte(active_frequency_idx, 1);
+  command[command_length++] = GetNthByte(active_frequency_idx, 0);
+  command[command_length++] = GetNthByte(passive_frequency_idx, 1);
+  command[command_length++] = GetNthByte(passive_frequency_idx, 0);
+
+  /** Creating the binary value */
+  uint16_t command_crc = CRCBitwise(command, command_length);
+  command[command_length++] = GetNthByte(command_crc, 1);
+  command[command_length++] = GetNthByte(command_crc, 0);
+  return command_length;
+}
+
+/**
+   * @brief parses the received data
+   * this is needed to save the active and passive station
+   * 
+   * @param string recived data
+   * @param len length of data
+   * @return true if data could be parsed
+   * @return false if data could not be parsed
+   */
+bool AR62xxDevice::ParseReceivedData(const char *string, size_t len)
+{
+  /** check if there is data */
+  if (string == NULL || len == 0)
+    return false;
+
+  size_t idx = 0;
+  uint16_t crc = 0;
+  static uint16_t rec_buf_len = 0;
+  int command_length = 0;
+#define REC_BUFSIZE 127
+  static uint8_t command[REC_BUFSIZE];
+
+  /** loop as long we have data */
+  while (idx < len)
+  {
+    /** check if we received a header*/
+    if ((uint8_t)string[idx] == 0xA5)
+      rec_buf_len = 0;
+
+    /** check if the buffer-size is reached */
+    if (rec_buf_len >= REC_BUFSIZE)
+      rec_buf_len = 0;
+    assert(rec_buf_len < REC_BUFSIZE);
+
+    /** write command in buffer */
+    command[rec_buf_len++] = (uint8_t)string[idx++];
+    if (rec_buf_len == 2)
```

Comment:
What is this magic number "2"?

Reply:
added comments about the magic number, should be clear now.